packages/amplication-gpt-gateway/src/conversationType/conversationType.service.ts

```diff
@@ -1,10 +1,41 @@
 import { Injectable } from "@nestjs/common";
+import { MessageInput } from "../dto/MessageInput";
+import { KafkaProducerService } from "../kafka/kafka.producer.service";
+import { MyMessageBrokerTopics } from "../kafka/topics";
 import { PrismaService } from "../prisma/prisma.service";
+import { TemplateService } from "../template/template.service";
 import { ConversationTypeServiceBase } from "./base/conversationType.service.base";
 
 @Injectable()
 export class ConversationTypeService extends ConversationTypeServiceBase {
-  constructor(protected readonly prisma: PrismaService) {
+  constructor(
+    protected readonly prisma: PrismaService,
+    protected readonly kafkaService: KafkaProducerService,
+    protected readonly templateService: TemplateService
+  ) {
     super(prisma);
   }
+
+  async startConversion(message: MessageInput): Promise<void> {
+    const { messageTypeId, params, requestUniqueId } = message;
+    const messageType = await this.prisma.conversationType.findUnique({
+      where: {
+        id: messageTypeId,
```

Comment:
we need to look for the conversation type by key and not by ID

```suggestion
        key: messageTypeKey,
```